packages/chatbot-server-mongodb-public/src/processors/generateResponseWithTools.ts

```diff
@@ -516,12 +760,25 @@ function formatMessageForReturnGeneration(
           m.content.forEach((c) => {
             if (c.type === "tool-result") {
               baseMessage.name = c.toolName;
-              const result = (c.result as Array<ToolResultPart | TextPart>)[0];
-              if (result.type === "text") {
-                baseMessage.content = result.text;
-              }
-              if (result.type === "tool-result") {
-                baseMessage.content = JSON.stringify(result.result);
+              if (
+                c.output.type === "content" &&
+                c.output.value[0].type === "text"
+              ) {
+                baseMessage.content = c.output.value[0].text;
+              } else if (
+                c.output &&
+                typeof c.output === "object" &&
+                "value" in c.output
+              ) {
+                baseMessage.content =
+                  typeof c.output.value === "string"
+                    ? c.output.value
+                    : JSON.stringify(c.output.value);
+              } else {
+                baseMessage.content =
+                  typeof c.output === "string"
+                    ? c.output
+                    : JSON.stringify(c.output);
```

Comment:
nit: I feel like this could be a helper, might make the logic a bit easier to read